api/include/opentelemetry/common/spin_lock_mutex.h

```diff
@@ -6,6 +6,17 @@
 
 #include "opentelemetry/version.h"
 
+#if defined(_MSC_VER)
+#  ifndef NOMINMAX
+#    define NOMINMAX
+#  endif
+#  include <Windows.h>
```

Comment:
C:\GH\otcpp\ext\include\opentelemetry\ext\http\client\curl\http_operation_curl.h(208):

Reply:
Sorry Tom, what do you mean by `C:\GH\otcpp\ext\include\opentelemetry\ext\http\client\curl\http_operation_curl.h(208)` ?

Reply:
Reason why I'm adding this is that Josh would need this in his PR that is failing. Related PR: #443

Reply:
> if the user could include windows.h before include our header file

Yeah, then it will break for those users. We most likely need to add some blurb in our documentation, that since our SDK uses range limits extensively in public API, that means we should tell users that they must revisit their usage of `min` and `max` macros, to make sure we are not breaking them on Windows.

Reply:
I tried fixing this where possible. But there are some spots, where our SDK defines class methods called `min` and `max` - that will surely break if our Windows customers don't define `NOMINMAX` themselves for their project.